src/ipc/ipc4/helper.c

```diff
@@ -391,7 +392,7 @@ __cold int ipc_pipeline_free(struct ipc *ipc, uint32_t comp_id)
 
 __cold static struct comp_buffer *ipc4_create_buffer(struct comp_dev *src, bool is_shared,
 						     uint32_t buf_size, uint32_t src_queue,
-						     uint32_t dst_queue)
+						     uint32_t dst_queue, struct k_heap *heap)
```

Comment:
The `comp_dev` structure contains the `mod` field that contains heap pointer. This new parameter is unnecessary.